### PR TITLE
Added Simkl

### PR DESCRIPTION
**Pull request by self-similarity**
Unfortunately also includes stuff from https://github.com/recloudstream/cloudstream/pull/538

### CODE REVIEW

File: app/src/main/java/com/lagradost/cloudstream3/syncproviders/SyncAPI.kt [hunk 1]
```diff
@@ -59,14 +60,24 @@ interface SyncAPI : OAuth2API {
         override var id: Int? = null,
     ) : SearchResponse
 
+    abstract class AbstractSyncStatus {
```
Comment (self-similarity) [inline app/src/main/java/com/lagradost/cloudstream3/syncproviders/SyncAPI.kt:63]: Required change for passing around data not present in the base class

**Comment by KillerDogeEmpire** [comment]
@self-similarity if it includes stuff from #538 close that PR

File: app/build.gradle.kts [hunk 2]
```diff
@@ -54,17 +55,27 @@ android {
...
         buildConfigField(
             "String",
             "BUILDDATE",
             "new java.text.SimpleDateFormat(\"yyyy-MM-dd HH:mm\").format(new java.util.Date(" + System.currentTimeMillis() + "L));"
         )
-
+        buildConfigField(
+            "String",
+            "SIMKL_CLIENT_ID",
+            "\"" + (System.getenv("SIMKL_CLIENT_ID") ?: localProperties["simkl.id"]) + "\""
```
Comment (C10udburst) [inline app/build.gradle.kts:72]: could you add these env vars to `.github/workflows/build_to_archive.yml` and `.github/workflows/prerelease.yml`?

Comment (self-similarity) [reply app/build.gradle.kts:72]: Added but untested :+1:

File: app/src/main/java/com/lagradost/cloudstream3/syncproviders/providers/SimklApi.kt [hunk 3]
```diff
@@ -0,0 +1,843 @@
...
+
+        val code = uri.getQueryParameter("code") ?: return false
+        val token = app.post(
+            "$mainUrl/oauth/token", json = TokenRequest(code)
+        ).parsedSafe<TokenResponse>() ?: return false
+
+        switchToNewAccount()
+        setKey(accountId, SIMKL_TOKEN_KEY, token.access_token)
+
+        val user = getUser()
```
Comment (LagradOst) [inline app/src/main/java/com/lagradost/cloudstream3/syncproviders/providers/SimklApi.kt:830]: if getUser throws an error switchToOldAccount will not get called, will lead to invalid state, add normalsafeapicall

**Review by LagradOst — APPROVED** [review]
just add one line and then LGTM 👍

**Pull request merged**